From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Arjun CR <arjuncr@ami.com>
Date: Sat, 15 Feb 2026 00:00:00 +0000
Subject: [PATCH] Remove -Wnull-dereference flag to fix boost warnings

The -Wnull-dereference flag is causing false positive warnings in
Boost ASIO headers (io_context.hpp, scheduler.hpp) when compiled
with newer GCC versions. These are false positives in library code
that we cannot fix. Removing this flag while keeping -Werror to
catch real issues in the project code.

Signed-off-by: Arjun CR <arjuncr@ami.com>
---
 CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1c50703..5f45c8e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,7 +16,6 @@ set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -lstdc++fs \
     -Wduplicated-cond \
     -Wduplicated-branches \
     -Wlogical-op \
-    -Wnull-dereference \
     -Wuseless-cast \
     -Wdouble-promotion \
     -Wformat=2 \
-- 
2.34.1

